src/config.rs

```diff
@@ -0,0 +1,158 @@
+use dashmap::DashMap;
+use ropey::Rope;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{
+    fs::read_to_string,
+    io::Error,
+    path::Path,
+    sync::{Arc, Mutex, MutexGuard, RwLock},
+};
+
+use crate::{htmx_tags::Tag, htmx_tree_sitter::LspFiles, init_hx::LangType, query_helper::Queries};
+use thiserror::Error;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct HtmxConfig {
+    pub lang: String,
+    pub template_ext: String,
+    pub templates: Vec<String>,
+    pub js_tags: Vec<String>,
+    pub backend_tags: Vec<String>,
+}
+
+impl HtmxConfig {
+    pub fn is_backend(&self, ext: &str) -> bool {
+        match self.lang.as_str() {
+            "rust" => ext == "rs",
```

Comment:
That's why I believe this feature should be a plugin instead so those who want to support a specific language would create their own implementation